src/os/bluestore/bluestore_types.h

```diff
@@ -1315,7 +1315,7 @@ struct sb_info_space_efficient_map_t {
 	  [](const sb_info_t& a, const uint64_t& b) {
 	    return a < b;
 	  });
-	if (it->get_sbid() == id) {
+        if (it != aux_items.end() && it->get_sbid() == id) {
```

Comment:
Just realized that this has been caused by improper bound specification in lower_bound call. See the code snippet above doing similar search over items container. It gets items.end() -1 as an end iterator and hence needs no check for it != items.end.
May be we should do the same here for the sake of code uniformity?

Reply:
Yeh, modifications should be uniform. So close these backports, open a new PR following 'items container' way in main branch then backport again?

Reply:
> Yeh, modifications should be uniform. So close these backports, open a new PR following 'items container' way in main branch then backport again?

yeah, this is a preferred way IMO.

Reply:
Open one as #56937, check plz.